### PR TITLE
Add copy as cURL button to Requestor

### PR DESCRIPTION
**Pull request by oscarvz**
Adds a 'copy as cURL' button to the bottom bar in Requestor. It's currently behind a feature flag still.

- Add `BottomToolbar` directory that houses its child components:
  - New: `CopyAsCurl`
  - Moved: `RequestBodyCombobox`
- Logic & user feedback when a button is disabled (based on method and/or unsupported types)

> [!note]
> I didn't bother with memoizing any of the logic implemented in `CopyAsCurl`, as every single interaction will result in a rerender - unless we memoize each prop individually, which introduces quite some overhead with negligible performance gains.

---
Can do:
https://github.com/user-attachments/assets/c2fe3a08-b39e-4b02-9ee6-b1bb9d9fe4f0

No can do:
https://github.com/user-attachments/assets/a19a7b31-7c59-414d-9a8c-02ceade66429

### CODE REVIEW

File: frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx
Comment (oscarvz) [inline frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx]: Depends on
- #168

Comment (oscarvz) [reply frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx]: Putting PR back into draft as above's PR is merged, resulting in duplicate headers being added to the cURL command.

File: frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx]: might wanna put the url in quotes, especially if there's a querystring attached, since most shells need a `?` to be wrapped in quotes!

to test locally, try making a GET request against a route with some query params attached

File: frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx:36]: i'll file an issue for this, but one thing that struck me is that with my silly "automatic content-type header" changes, we attach a content-type to GET requests, which isn't necessary.

but yeah. that's an upstream issue, i can fix it

File: frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/utils.ts
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/utils.ts]: so instead of constructing this manually, you could also make use of `URLSearchParams` and that would help you get encoding for free. i should've thought of this when we talked yesterday but it just hit me 

```ts
import { FormDataParameter } from "../../FormDataForm/types";

export function formDataToUrlEncoded(formDataParams: FormDataParameter[]) {
  const urlEncodedFormData = new URLSearchParams();

  formDataParams.forEach((field) => {
    if (field.enabled && field.value.type === "text") {
      urlEncodedFormData.append(field.key, field.value.value);
    }
  });

  return urlEncodedFormData.toString();
}
```

Comment (brettimus) [reply frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/utils.ts]: then in your switch case just call `formDataToUrlEncoded(body.value)`

File: frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx
Comment (oscarvz) [reply frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx]: Good one! Will add

File: frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/utils.ts
Comment (oscarvz) [reply frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/utils.ts]: This indeed is a nicer way to approach it. Will add!

File: frontend/src/pages/RequestorPage/RequestPanel/RequestPanel.tsx
Comment (brettimus) [inline frontend/src/pages/RequestorPage/RequestPanel/RequestPanel.tsx]: without the `overflow-hidden` here we don't get scrollability for the inner contents when they overflow (many request params, long req body)

Comment (brettimus) [reply frontend/src/pages/RequestorPage/RequestPanel/RequestPanel.tsx]: then once we do have a scrollable container, might need to add some bottom padding: 
image

**Review by brettimus — APPROVED** [review]
pre-approving, but we gotta fix the issues with vertically scrolling through overflowing content in the request panel before smashing that merge button. left a comment on the relevant code

**Pull request merged**
